opcua/server/address_space.py

```diff
@@ -108,7 +108,8 @@ def _suitable_reftype(self, ref1, ref2, subtypes):
         """
         if ref1.Identifier == ref2.Identifier:
             return True
-        if not subtypes and ref2.Identifier == ua.ObjectIds.HasSubtype:
+        #TODO: Please check the changes here
+        elif not subtypes:
```

Comment:
did the original implementation break something? This is copied from C++ and I kind of remember we had issues here. But removing the last test as you did seems to make sense... if it does not break tests we can try with your version...